### PR TITLE
Use actual API route for tracks list

### LINKED ISSUE #187: This URL is still points to the test route
@kntsoriano Can you change it pls 🙂 

https://github.com/exercism/v3-website/blob/7bcf1cfa470359e680901dfd4d01822b2cd2b8fa/app/views/tracks/index/authenticated.html.haml#L2

### PR DESCRIPTION
**Pull request by kntsoriano**
Closes https://github.com/exercism/v3-website/issues/187.

### CODE REVIEW

File: app/helpers/view_components/student/tracks_list.rb [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
         @data = data
         @request = request
       end
 
       def to_s
         react_component("student-tracks-list", {
-                          request: request.deep_merge({ options: { initialData: data } }),
+                          request: request.deep_merge({
+                                                        options: { initialData: data },
+                                                        query: { auth_token: user.auth_tokens.first.to_s }
```
Comment (kntsoriano) [inline app/helpers/view_components/student/tracks_list.rb:15]: We now need to pass the auth token in order for the API to respond.

File: app/helpers/view_components/student/tracks_list.rb [hunk 2]
```diff
@@ -1,14 +1,18 @@
 module ViewComponents
   module Student
     class TracksList < ViewComponent
-      def initialize(data, request = default_request)
+      def initialize(user, data, request = default_request)
```
Comment (iHiD) [inline app/helpers/view_components/student/tracks_list.rb:5]:
```suggestion
      TODO: Remove `user` and its usage here once API supports session requests
      def initialize(user, data, request = default_request)
```

File: app/helpers/view_components/student/tracks_list.rb [hunk 3]
```diff
@@ -1,14 +1,19 @@
 module ViewComponents
   module Student
     class TracksList < ViewComponent
-      def initialize(data, request = default_request)
+      TODO: Remove `user` and its usage here once API supports session requests
```
Comment (iHiD) [inline app/helpers/view_components/student/tracks_list.rb]:
```suggestion
      # TODO: Remove `user` and its usage here once API supports session requests
```

File: app/views/tracks/index/authenticated.html.haml [hunk 4]
```diff
@@ -1,2 +1,2 @@
 #page-tracks
-  = ViewComponents::Student::TracksList.new(@tracks_data, { endpoint: tracks_test_components_student_tracks_list_path, query: { state: params[:state] }, options: { retry: params[:state] == "Loading", initialStale: true } })
+  = ViewComponents::Student::TracksList.new(current_user, @tracks_data)
```
Comment (iHiD) [inline app/views/tracks/index/authenticated.html.haml:2]: Do we not need the `params[:state]` thing? Is that just for testing?

File: app/helpers/view_components/student/tracks_list.rb (hunk 1 above)
Comment (iHiD) [reply app/helpers/view_components/student/tracks_list.rb:15]: Thanks. Yeah - this is temporary until the API supports sessions. I've added a TODO to remove it again in the future that in this PR.

**Review by iHiD — APPROVED** [review]

File: app/views/tracks/index/authenticated.html.haml (hunk 4 above)
Comment (kntsoriano) [reply app/views/tracks/index/authenticated.html.haml:2]: Just for testing.

**Review by ErikSchierboom — APPROVED** [review]
LGTM

**Pull request merged**